### PR TITLE
SIP-118 Conditional Fee Reclamation

### PR DESCRIPTION
**Pull request by i-stam**
implementation of https://github.com/Synthetixio/SIPs/pull/403

### CODE REVIEW

File: publish/src/commands/deploy-ovm-pair.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ const deployInstance = async ({ useOvm, privateKey, l1ProviderUrl, l2ProviderUrl
 		methodCallGasLimit: '3500000',
 		contractDeploymentGasLimit: useOvm ? OVM_MAX_GAS_LIMIT : '9500000',
 		privateKey,
-		ignoreCustomParameters: true,
+		ignoreCustomParameters: false,
```
Comment (justinjmoses) [inline publish/src/commands/deploy-ovm-pair.js:61]: @i-stam is this necessary if instead we have the local-ovm tests simply check what `waitingPeriodSecs` is on-chain and test based on that?

Comment (i-stam) [reply publish/src/commands/deploy-ovm-pair.js:61]: not necessary in this specific PR, we will address this when we update the ovm prod tests

File: test/contracts/Exchanger.spec.js [hunk 2]
```diff
@@ -192,6 +194,141 @@ contract('Exchanger (spec tests)', async accounts => {
 		});
 	});
 
+	describe('When the waiting period is set to 0', () => {
+		let initialWaitingPeriod;
+
+		beforeEach(async () => {
+			initialWaitingPeriod = await systemSettings.waitingPeriodSecs();
+			await systemSettings.setWaitingPeriodSecs('0', { from: owner });
```
Comment (justinjmoses) [inline test/contracts/Exchanger.spec.js:202]: this isn't necessary to checj here but rather in `SystemSettings.js`

File: test/contracts/Exchanger.spec.js [hunk 3]
```diff
@@ -192,6 +194,141 @@ contract('Exchanger (spec tests)', async accounts => {
...
+				assert.bnEqual(await sETHContract.balanceOf(owner), '0');
+			});
+
+			describe('When the waiting period is switched on again', () => {
+				beforeEach(async () => {
+					await systemSettings.setWaitingPeriodSecs(initialWaitingPeriod, { from: owner });
+				});
+
+				it('is set correctly', async () => {
+					assert.bnEqual(await systemSettings.waitingPeriodSecs(), initialWaitingPeriod);
```
Comment (justinjmoses) [inline test/contracts/Exchanger.spec.js:241]: as before, this isn't necessary to check as the logic belongs to SystemSettings

**Review by justinjmoses — APPROVED** [review]
💯 Nice work - just a note about a test or two.

File: test/contracts/Exchanger.spec.js (hunk 3 above)
Comment (i-stam) [reply test/contracts/Exchanger.spec.js:241]: Hmmm, I see, I just added it to make it more explicit that the change has taken effect.

File: contracts/Exchanger.sol [hunk 4]
```diff
@@ -587,15 +587,18 @@ contract Exchanger is Owned, MixinSystemSettings, IExchanger {
...
-        // persist the exchange information for the dest key
-        appendExchange(
-            destinationAddress,
-            sourceCurrencyKey,
-            sourceAmountAfterSettlement,
-            destinationCurrencyKey,
-            amountReceived,
-            exchangeFeeRate
-        );
+        // iff the waiting period is gt 0
```
Comment (gberg1) [inline contracts/Exchanger.sol:590]: @i-stam extra f in iff

Comment (i-stam) [reply contracts/Exchanger.sol:590]: ah, this is an abbreviation of `if and only if` i.e.` <=>`

**Pull request merged**